src/NuGet.Core/NuGet.Commands/SourcesCommands/SourcesRunner.cs

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Linq;
+using NuGet.Common;
+
+namespace NuGet.Commands
+{
+    /// <summary>
+    /// Shared code to run the "sources" command from the command line projects
+    /// </summary>
+    public static class SourcesRunner
+    {
+        public static void Run(SourcesArgs args)
+        {
+            switch (args.Action)
+            {
+                case SourcesAction.List:
+                    switch (args.Format)
+                    {
+                        case SourcesListFormat.Short:
+                            PrintRegisteredSourcesShort(args);
+                            break;
+                        case SourcesListFormat.Detailed:
+                        default:
+                            PrintRegisteredSourcesDetailed(args);
+                            break;
+                    }
+                    break;
+                case SourcesAction.Add:
+                    AddNewSource(args);
+                    break;
+                case SourcesAction.Remove:
+                    RemoveSource(args);
+                    break;
+                case SourcesAction.Enable:
+                    EnableOrDisableSource(args, enabled: true);
+                    break;
+                case SourcesAction.Disable:
+                    EnableOrDisableSource(args, enabled: false);
+                    break;
+                case SourcesAction.Update:
+                    UpdatePackageSource(args);
+                    break;
+            }
+        }
+
+        private static void EnableOrDisableSource(SourcesArgs args, bool enabled)
+        {
+            if (string.IsNullOrEmpty(args.Name))
+            {
+                throw new CommandLineException(Strings.SourcesCommandNameRequired);
```

Comment:
Why not use ArgumentException?

Reply:
Discussed offline. 

Whether we call this CommandLineException or something smarter like LocalizedException can be decided in the dotnet.exe side implementation of this.

Reply:
fwiw, looking into our code...there's nothing that relies on this being a commandlineexception.